src/main/java/dev/binclub/javaception/runtime/ExecutionEngine.java

```diff
@@ -22,20 +32,696 @@ public static Object invokeMethodObj(InstanceOop instance, MethodInfo method, Ob
+				if (Double.isNaN(d) || Double.isNaN(d2)) {
+					if (opcode == DCMPL) {
+						methodContext.push(-1);
+					} else {
+						methodContext.push(1);
+					}
+				} else {
+					if (methodContext.pop().equals(methodContext.pop())) {
+						methodContext.push(0);
+					} else {
+						if ((double) methodContext.pop() < (double) methodContext.pop()) {
+							methodContext.push(1);
+						} else {
+							methodContext.push(-1);
+						}
+						
+					}
+				}
+				break;
+			case IFEQ:
+				if ((int) methodContext.pop() == 0) {
+					branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				}
+				break;
+			case IFNE:
+				if ((int) methodContext.pop() != 0) {
+					branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				}
+				break;
+			case IFLT:
+				if ((int) methodContext.pop() < 0) {
+					branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				}
+				break;
+			case IFGE:
+				if ((int) methodContext.pop() >= 0) {
+					branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				}
+				break;
+			case IFGT:
+				if ((int) methodContext.pop() > 0) {
+					branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				}
+				break;
+			case IFLE:
+				if ((int) methodContext.pop() <= 0) {
+					branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				}
+				break;
+			case IF_ICMPEQ:
+				if ((int) methodContext.pop() == (int) methodContext.pop()) {
+					branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				}
+				break;
+			case IF_ICMPNE:
+				if ((int) methodContext.pop() != (int) methodContext.pop()) {
+					branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				}
+				break;
+			case IF_ICMPLT:
+				if ((int) methodContext.pop() > (int) methodContext.pop()) {
+					branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				}
+				break;
+			case IF_ICMPGE:
+				if ((int) methodContext.pop() <= (int) methodContext.pop()) {
+					branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				}
+				break;
+			case IF_ICMPGT:
+				if ((int) methodContext.pop() < (int) methodContext.pop()) {
+					branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				}
+				break;
+			case IF_ICMPLE:
+				if ((int) methodContext.pop() >= (int) methodContext.pop()) {
+					branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				}
+				break;
+			case IF_ACMPEQ:
+				if (methodContext.pop() == methodContext.pop()) {
+					branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				}
+				break;
+			case IF_ACMPNE:
+				if (methodContext.pop() != methodContext.pop()) {
+					branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				}
+				break;
+			case GOTO:
+				branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				break;
+			case JSR:
```

Comment:
I think its best if we just dont support JSR/JSR_W/RET

Reply:
doesnt hurt too much to support it and if we don't could potentially be used against us in the future